### PR TITLE
Simulation

### PR DESCRIPTION
**Pull request by BurntVal**

### CODE REVIEW

File: packages/signers/src/signers/utils/client.ts [hunk 1]
```diff
@@ -135,10 +177,122 @@ export class AAClient extends SigningCosmWasmClient {
...
+
+    const chainId = await this.getChainId();
+
+    if (/testnet/.test(chainId)) {
+      defaultFee = { amount: [{ amount: "0", denom: "uxion" }], gas: gas };
+    } else {
+      defaultFee = { amount: calculatedFee.amount, gas: gas };
+    }
+
+    const tx = await this.sign(sender, [addMsg], fee || defaultFee, memo);
```
Comment (BurntNerve) [inline packages/signers/src/signers/utils/client.ts:208]: If this is repeated should we just make this a function?

Comment (BurntVal) [reply packages/signers/src/signers/utils/client.ts]: Good point. Fixed

**Review by BurntNerve — APPROVED** [review]
Looks good!

File: packages/signers/src/signers/utils/client.ts [hunk 2]
```diff
@@ -135,10 +195,97 @@ export class AAClient extends SigningCosmWasmClient {
...
+      txBytes: TxRaw.encode(tx).finish(),
+    });
+
+    const { gasInfo } = await queryService.Simulate(request);
+
+    if (!gasInfo) {
+      throw new Error("No gas info returned");
+    }
+
+    return Uint53.fromString(gasInfo.gasUsed.toString()).toNumber();
```
Comment (ash-burnt) [inline packages/signers/src/signers/utils/client.ts:286]: Why UInt53?

**Review by ash-burnt — APPROVED** [review]

**Pull request merged**